cmdutil/debug/debug.go

```diff
@@ -31,25 +33,74 @@ import (
 // Connect to the debug server with gops:
 //
 //	gops stack localhost:PORT
-func New(l logrus.FieldLogger, port int) *Server {
-	return &Server{
+//
+// Connect to the pprof server with pprof.port when pprof enabled
+func New(l logrus.FieldLogger, config Config) *Server {
+	server := &Server{
 		logger: l,
-		addr:   fmt.Sprintf("127.0.0.1:%d", port),
+		addr:   fmt.Sprintf("127.0.0.1:%d", config.Port),
 		done:   make(chan struct{}),
 	}
+	if config.Enabled {
+		server.pprof = NewPProfServer(l, &config.PProf)
+	}
+	return server
 }
 
 // Server wraps a gops server for easy use with oklog/group.
 type Server struct {
 	logger logrus.FieldLogger
 	addr   string
 	done   chan struct{}
+	pprof  *PProfServer
 }
 
 // Run starts the debug server.
 //
-// It implements oklog group's runFn.
+// It implements oklog group's runFn and pprof.
 func (s *Server) Run() error {
+
+	var wg sync.WaitGroup
+	gopsErrChan := make(chan error, 1)
+	pprofErrChan := make(chan error, 1)
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		gopsErrChan <- s.RunGOPS()
```

Comment:
Why should this be a channel? You can simply do:
```
gopsErr = s.RunGOPS()
```

The call to `wg.Wait()` on line 81 will mean that you block until both services have stopped running. In fact, you should probably catch the error here and log it if one of the services has stopped running. You could end up with the pprof server exiting in error and gops server running indefinitely, with no indication as to why or that this happened.

Reply:
Updated it